subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/ivy/DefaultIvyModuleResolveMetadata.java

```diff
@@ -88,6 +91,30 @@ private DefaultIvyModuleResolveMetadata(DefaultIvyModuleResolveMetadata metadata
         // Cached state is not copied, since dependency inputs are different.
     }
 
+    @Override
+    protected Optional<ImmutableList<? extends ConfigurationMetadata>> maybeDeriveVariants() {
+        if (derivedVariants == null) {
+            for (String potentialVariantName : getConfigurationNames()) {
+                if (getVariantMetadataRules().willApplyVariantAttributeRules(potentialVariantName)) {
+                    derivedVariants = Optional.of(allConfigurationsWithAttributes());
```

Comment:
That code construct is fishy, you loop again on `getConfigurationNames()` in the called method. It might be better to loop only once and check if the builder had anything added to it.